lib/spack/spack/modules/common.py

```diff
@@ -403,8 +403,8 @@ def get_module(module_type, spec, get_full_path, module_set_name="default", requ
         writer = spack.modules.module_types[module_type](spec, module_set_name)
         if not os.path.isfile(writer.layout.filename):
             if not writer.conf.excluded:
-                err_msg = "No module available for package {0} at {1}".format(
-                    spec, writer.layout.filename
+                err_msg = "The module for package /{0} should be at {1} but it does not exist!".format(
+                    spec.dag_hash(), writer.layout.filename
```

Comment:
Can we maintain the spec name and version along with the hash? That shouldn't result in an extremely long message, and would improve readability.

Reply:
Also, now we're allowed to use f-strings, since Spack supports running Python 3.6+

Reply:
Done, assuming the correct syntax is `spec.name` and `spec.version`

Reply:
Tested:
```
==> Error: The module for package gcc@9.4.0/occi2durd5 should be at /modules/spack-test-hierarchy/share/spack/modules/linux-ubuntu20.04-x86_64/gcc-9.4.0-gcc-9.4.0-occi2du but it does not exist!
```